stdlib/public/runtime/ReflectionMirror.cpp

```diff
@@ -1092,3 +1092,28 @@ id swift_reflectionMirror_quickLookObject(OpaqueValue *value, const Metadata *T)
 #endif
 
 #endif  // SWIFT_ENABLE_REFLECTION
+
+#include "ImageInspection.h"
+
+SWIFT_CC(swift) SWIFT_RUNTIME_STDLIB_API
+const char *swift_keypath_dladdr(void *address) {
+    SymbolInfo info;
+    if (lookupSymbol(address, &info) == 0) {
+        return 0;
+    } else {
+        return info.symbolName.get();
+    }
+}
+
+SWIFT_CC(swift) SWIFT_RUNTIME_STDLIB_API
+const char *swift_keypathSourceString(char *name) {
+    size_t length = strlen(name);
+    std::string mangledName = keypathSourceString(name, length);
+    if (mangledName == "") {
+        return 0;
+    } else {
+        char *rtnValue = new char[mangledName.length() + 1];
```

Comment:
I wonder if this function can take a closure that passes `mangledName.c_str()` here instead. (I'm somewhat nervous about using `new` and not having a related `delete`, even though it has `.deallocate()` on the Swift side).

Reply:
I'll look into this, but idk if that's possible. I'm guessing I'm limited to C language features in this context, and I don't think Swift allows captures in a convention(c) closure.

Reply:
With that said, I share your concerns about the allocation with the dealloc far away in code written in another language entirely.

Reply:
This should be resolved. You can't capture variables in a c convention closure, so I've returned a class instead using  `Unmanaged`. If there's a better way to guarantee that a `String` outlives the scope of the closure let me know and I'll switch.